DEFEDIT-1113 Better support for shortcut keys

ragnard-king opened the pull request:
Fixes https://github.com/defold/editor2-issues/issues/898, https://github.com/defold/editor2-issues/issues/1126, https://github.com/defold/editor2-issues/issues/425, https://github.com/defold/editor2-issues/issues/491

### User facing changes

- We now support a greater range of shortcut keys, including single
  key shortcuts (F for frame etc.)
- Aligned some shortcuts with editor 1, for example, space now toggles
  animations in collection/game object view, palette in tile map view. W, E, R selects move/rotate/scale tool.


### Technical changes

- We no longer use JavaFX `MenuItems` bound to accelerators to dispatch
  shortcut key events, meaning we are no longer subject to the many limitations of that approach.
- Introduced `editor.keymap` namespace for defining mappings from
  shortcut keys ("Ctrl+A") to editor commands (`:select-all`) and
  installing scene accelerators based on such definitions.
- Separated the definition of menus and shortcuts, and extracted all
  existing shortcuts to a `default-key-bindings` list in
  `editor.keymap`.

ragnard-king committed:
DEFEDIT-1113 Better shortcuts

* User facing changes

- We now support a greater range of shortcut keys, including single
  key shortcuts (F for frame etc.)
- Aligned some shortcuts with editor 1, for example, space now toggles
  animations in collection/game object view, palette in tile map view.

* Technical changes

- We no longer use the JavaFX accelerator mechanism to dispatch
  shortcut key events, meaning we are no longer subjetc to its many
  limitations.
- Introduced `editor.keymap` namespace for defining mappings from
  shortcut keys ("Ctrl+A") to editor commands (`:select-all`) and
  matching key events to commands, based on such definitions.
- Separated the definition of menus and shortcuts, and extracted all
  existing shortcuts to a `default-key-bindings` list in
  `editor.keymap`.

matgis-king reviewed (changes requested):
Really liking this! Found a few issues we might want to address before merging, though.

ragnard-king committed:
DEFEDIT-1113 Sort default-key-bindings by shortcut key.

ragnard-king committed:
DEFEDIT-1113 Remove redundant :toggle shortcut

ragnard-king committed:
DEFEDIT-1113 Add docstring to `remove-shortcut-key-overlaps`.

ragnard-king committed:
DEFEDIT-1113 Use correct command for "Show palette"

ragnard-king committed:
DEFEDIT-1113 Clearer keys for key-combo-data

ragnard-king committed:
DEFEDIT-1113 Revert `show-dialog-stage`

Fixes macOS issue where incorrect commands are available when a dialog
is open.

ragnard-king committed:
DEFEDIT-1113 Allow duplicate shortcuts for some select commands

Workaround until we have a saner way of organising shortcuts for
different editors/resource types.

ragnard-king committed:
DEFEDIT-1113 Introduce mechanism to override unwanted key bindings

- some EventDispatch trickery so we can choose to ignore some default
  key bindings for any control
- ignore `Space` on `TreeView` in outline (which toggles selection) so
  that we instead can use space for scene play and toggle tile map palette.

ragnard-king committed:
DEFEDIT-1113 Fix failing tests

ragnard-king committed:
DEFEDIT-1113 Fix renaming mistake
